jujugui/static/gui/src/app/store/env/go.js

```diff
@@ -729,39 +735,130 @@ YUI.add('juju-env-go', function(Y) {
-          provider: result.ProviderType,
-          uuid: result.UUID,
-          serverUuid: result.ServerUUID,
-          ownerTag: result['owner-tag']
+        var models = results.map(function(result, index) {
+          err = result.error && result.error.Message;
+          if (err) {
+            return {tag: tags[index], err: err};
+          }
+          result = result.result;
+          return {
+            tag: tags[index],
+            name: result.Name,
+            series: result.DefaultSeries,
+            provider: result.ProviderType,
+            uuid: result.UUID,
+            serverUuid: result.ServerUUID,
+            ownerTag: result.OwnerTag,
+            life: result.Life,
+            isAlive: result.Life === 'alive',
+            isAdmin: result.UUID === result.ServerUUID
+          };
         });
-      }.bind(this, callback, modelTag);
+        userCallback({models: models});
+      }.bind(this, callback, tags);
 
       // Send the API request.
+      var entities = tags.map(function(tag) {
+        return {Tag: tag};
+      });
       this._send_rpc({
         Type: 'ModelManager',
         Request: 'ModelInfo',
-        Params: {Entities: [{Tag: modelTag}]}
+        Params: {Entities: entities}
       }, handler);
     },
 
     /**
+      Return detailed information about Juju models available for current user.
+      Under the hood, this call leverages the ModelManager ListModels and
+      ModelInfo endpoints.
+
+      @method listModelsWithInfo
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive an object with an "err"
+        attribute containing a string describing the problem (if an error
+        occurred). Otherwise, if everything went well, it will receive an
+        object with a "models" attribute containing an array of model info,
+        each one with the following fields :
+        - tag: the original Juju model tag;
+        - name: the model name, like "admin" or "mymodel";
+        - series: the model default series, like "trusty" or "xenial";
+        - provider: the provider type, like "lxd" or "aws";
+        - uuid: the model unique identifier;
+        - serverUuid: the corresponding controller unique identifier;
+        - ownerTag: the Juju tag of the user owning the model;
+        - life: the lifecycle status of the model: "alive", "dying" or "dead";
+        - isAlive: whether the model is alive or dying/dead;
+        - isAdmin: whether the model is an admin model;
+        - lastConnection: the date of the last connection as a string, e.g.:
+          '2015-09-24T10:08:50Z' or null if the model was never connected to;
+        - err: a message describing a specific model error, or undefined.
+      @return {undefined} Sends a message to the server only.
+    */
+    listModelsWithInfo: function(callback) {
+      // Ensure we always have a callback.
+      if (!callback) {
```

Comment:
For my own curiosity… I don't recall seeing this pattern in other functions that take callbacks. Is it something that should be more widespread?